scripts/src/primazactl/parser.py

```diff
@@ -0,0 +1,35 @@
+import argparse
+import sys
+from primazactl.primazamain import parser as mainparser
+from primazactl.primazaworker import parser as workerparser
+
+
+class PrimazactlParser(argparse.ArgumentParser):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+
+    def error(self, message):
+        sys.stderr.write('error: %s\n' % message)
+        sys.exit(2)
+
+
+def build_parser() -> argparse.ArgumentParser:
+    parser = PrimazactlParser(
+        prog='primazactl',
+        description='Configure and install primaza and primaza '
+                    'worker on clusters',
+        epilog="Brought to you by the RedHat app-services team.")
+
+    base_subparser = argparse.ArgumentParser(add_help=False)
+    base_subparser.add_argument(
+        "-x", "--verbose",
+        dest="verbose",
+        required=False,
+        action=argparse.BooleanOptionalAction,
+        help="Set for verbose output")
```

Comment:
This flag is not handled anywhere so does not work. I guess it needs processing in each command so in 3 separate places but not sure. To process is you need to call logger.set_verbose(verbose value).
Als it would be better if action did not all have to make the same call.

Reply:
I agree. Do you think updating the `main()`  as follows may work? However, I'd propose to do this a separate PR.
Also, we should configure the [action field](https://docs.python.org/3/library/argparse.html#action) of the argument as `count` 

```diff
def main():
    p = parser.build_parser()

    try:
        args = p.parse_args()
+       if hasattr(args, "verbose"):
+          logger.set_verbose(args.verbose)

        if not hasattr(args, "func"):
            p.print_help()
        else:
            args.func(args)
    except (argparse.ArgumentError, ValidationError) as err:
        p.error(err)
    except Exception as err:
        sys.stderr.write("error: %s\n" % err)
        sys.exit(1)
```

Reply:
Yes, that would work, but may be better in a different function which porovides a pattern for processing generic arguments which cover all commands even though at the moment we have only one.